Content.Client/GameObjects/Components/PDA/PDABoundUserInterface.cs

```diff
@@ -0,0 +1,186 @@
+using Content.Client.Utility;
+using Content.Shared.GameObjects.Components.PDA;
+using Robust.Client.GameObjects.Components.UserInterface;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Shared.GameObjects.Components.UserInterface;
+using Robust.Shared.Localization;
+using Robust.Shared.Maths;
+
+namespace Content.Client.GameObjects.Components.PDA
+{
+    public class PDABoundUserInterface : BoundUserInterface
+    {
+
+        private PDAMenu _menu;
+        private ClientUserInterfaceComponent Owner;
+        public PDABoundUserInterface(ClientUserInterfaceComponent owner, object uiKey) : base(owner, uiKey)
+        {
+            Owner = owner;
+        }
+
+        protected override void Open()
+        {
+            base.Open();
+            _menu = new PDAMenu(this);
+            _menu.OpenToLeft();
+            _menu.OnClose += Close;
+            _menu.FlashLightToggleButton.OnToggled += args =>
+            {
+                SendMessage(new PDAToggleFlashlightMessage());
+
+            };
+
+            _menu.EjectIDButton.OnPressed += args =>
+            {
+                SendMessage(new PDAEjectIDMessage());
+            };
+        }
+
+        protected override void UpdateState(BoundUserInterfaceState state)
+        {
+            base.UpdateState(state);
+            if (!(state is PDAUpdateUserInterfaceState cstate))
```

Comment:
Should this be an assertion, rather than a silent failure? Don't know how the other code has been handling it.

Reply:
Yeah. While drafting I just copy paste from other components to avoid typing SO MUCH boilerplate. This is how it's done in a few places but I'll assert it here instead.